Feature/terraform implementation fix (#60)
Azure/aca-landing-zone-accelerator: scenarios/aca-internal/terraform/modules/06-application-gateway/README.md

# Expose the workload through Application Gateway

The [workload that was deployed](../05-hello-world-sample-app/README.md) in the prior step needs to be exposed through a controlled gateway to reachable. Here, you'll deploy and configure Azure Application Gateway to serve that role.

## Expected results

The "Hello World" container app is exposed through Application Gateway, including with a TLS certificate that is stored in Key Vault. The cert is pre-generated.

![A picture of the configuration of Application Gateway.](./media/application-gateway.png)

### Resources

- Application Gateway with public IP
- SSL Certificate in Key Vault of the supporting services
- User Assigned Managed Identity for Application Gateway to access the secret in the Key Vault

## Steps

If you want to use remote storage, uncomment the backend block in the `providers.tf` file and provide the information for your Azure Storage Account. 

> 1. Deploy and configure Application Gateway with TLS

Reviewer: ```suggestion
1. Deploy and configure Application Gateway with TLS

```

```bash

> terraform init

Reviewer: Looks like this isn't indented properly, you can see the command below for reference

terraform plan -out tfplan

> terraform apply tfplan 

Reviewer: ditto

- SSL Certificate in Key Vault of the supporting services
- User Assigned Managed Identity for Application Gateway to access the secret in the Key Vault

#### Configure Terraform remote state

To configure your Terraform deployment to use the newly provisioned storage account and container, edit the [`./providers.tf`](./providers.tf) file at lines 11-13 as below:

```hcl

>   backend "azurerm" {
>     resource_group_name  = "<REPLACE with $RESOURCE_GROUP_NAME>"
>     storage_account_name = "<REPLACE with $STORAGE_ACCOUNT_NAME>"
>     container_name       = "tfstate"
>     key                  = "myapp/container-apps-environment.tfstate"
>   }

Reviewer: ```suggestion
backend "azurerm" {
  resource_group_name  = "<REPLACE with $RESOURCE_GROUP_NAME>"
  storage_account_name = "<REPLACE with $STORAGE_ACCOUNT_NAME>"
  container_name       = "tfstate"
  key                  = "myapp/container-apps-environment.tfstate"
}
```

```

* `resource_group_name`: Name of the Azure Resource Group that the storage account resides in.
* `storage_account_name`: Name of the Azure Storage Account to be used to hold remote state.
* `container_name`: Name of the Azure Storage Account Blob Container to store remote state.
* `key`: Path and filename for the remote state file to be placed in the Storage Account Container. If the state file does not exist in this path, Terraform will automatically generate one for you.
  
## Steps

> 1. Navigate to the Terraform module for the hub. 

Reviewer: ```suggestion
1. Navigate to the Terraform module for the Application Gateway configuration.
```

   
   ```bash

>    cd modules/06-application-gateway

Reviewer: ```suggestion
   cd ../06-application-gateway
```
